src/index.ts

```diff
@@ -112,8 +113,9 @@ export const useCourseSearch = (
   clearSearch: () => void,
   aggregations: Aggregations,
   loaded: boolean,
-  searchPageSize: number,
-  history: HHistory
+  searchPageSize: number | null,
+  history: HHistory,
+  getPageSizeFromUIParam: ((ui: string | null) => number) | null = null
```

Comment:
**Requested change:** Could we have just a single size parameter? The current usage is potentially confusing imo...if both are passed, which takes precedence? To avoid that question, you could pass `null, history, function` like in https://github.com/mitodl/ocw-hugo-themes/pull/953/files#diff-b681b498395009bcd9a1b7c4c214db07e6997fd551aec850c8a026447a7e70f2R163, but that seems a little awkward.

What would you think of:


```suggestion
  pageSize: number | GetSearchPageSize,
  history: HHistory,
```
where
```ts
type GetSearchPageSize = (ui: string) => number
```

_I mildly recommend an object for the `GetSearchPageSize` parameter rather than just `ui => number`, but I don't think it matters much._